dimmer.el

```diff
@@ -132,26 +132,18 @@ color is brighter then we return t, else nil."
 (defun dimmer-compute-rgb (c pct invert)
   "Computes the color C when dimmed by percentage PCT.
 When INVERT is true, make the value brighter rather than darker."
-  (let ((rgb (color-name-to-rgb c)))
-    (when rgb
-      (apply 'color-rgb-to-hex
-             (mapcar
-              (if invert
-                  (lambda (x) (- 1.0 (* (- 1.0 x) (- 1.0 pct))))
-                (lambda (x) (* x (- 1.0 pct))))
-              rgb)))))
+  (let ((percent (* 100 pct (if invert 1 -1))))
+    (color-desaturate-name (color-lighten-name c percent) (abs percent))))
```

Comment:
Can you wrap this in a `(when (color-defined-p c) ...` 

Because `color-lighten-name` and `color-desaturate-name` can throw a `wrong-number-of-arguments` if the color c isn't defined.  It's okay for this function to return `nil`.

Reply:
Also, I'm experimenting a little with the desaturation and I'm not completely happy with it. It may be a matter of preference but let me explain.

One problem is that the saturation effect is a lot more sensitive to the "percent" parameter than the lightness adjustment.  The theme I use (Zenburn) happens to be relatively desaturated by design already, and what ends up happening for me is faces becoming nearly or completely gray when the window is deselected, and that defeats the benefits of syntax highlighting for me.

The second problem, is relatively dim foreground colors can be darkened so that they "flip" darker than the background.  In zenburn, my clojure comments go from sage green to a gray that is a slightly darker than the background. Aesthetically I'd prefer not to have that happen.  And granted it could happen even in the current implementation. I'm not sure why, but combining the desat and the darken together makes this happen in my theme.

I'm tempted to suggest separating the dimming fraction from the desaturation fraction, so that people like me can reduce (or defeat) the desaturation effect.

Reply:
Yeah, we could also just skip the desaturation step: even the `color-lighten-name` function does a better job than the previous code at adjusting the perceived brightness. The issue I have with brightening light themes is that some colours then "pop" a little more, so the dimming effect isn't very uniform. I haven't tried *just* desaturating the colours, though. I feel like there might always be some themes (e.g. monochrome themes) for which this library won't always be able to do the right thing, so you could provide a custom variable which names a function to be used for adjusting the colours. In the extreme case, then, theme authors could even provide that function along with their themes, since themes can (by design) include any custom variables.

Reply:
P.S. I added the `color-defined-p` check.

Reply:
Pushed a few changes, including halving the percentage for the desaturation step.